src/lib.rs

```diff
@@ -1,7 +1,38 @@
+extern crate regex;
+
+mod links;
+
+#[derive(Debug)]
+pub struct MessageMeta {
+    pub message: String,
+    pub links: Vec<links::Link>,
+}
+
+pub fn parse<T: ToString>(message: T) -> MessageMeta {
+    let msg = message.to_string();
+    let links = links::parse(msg.clone());
+
+    MessageMeta { message: msg.to_string(), links }
```

Comment:
Masz dwa razy `.to_string()` - tutaj i wyżej przy `let`. Gdybyś zmienił nazwę z `msg` na `message` to mógłbyś napisać po prostu: `MessageMeta { message, links }`

Reply:
prawda, można to fajnie odwrócić 👍